### ISSUE TITLE
68 Portland - SAB Latency

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1496]
In GitLab by @kdjstudios on Jun 7, 2019, 09:22

**Submitted by:** "Grant Crymes" <grant.crymes@answernet.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/8513066 - CLOSED

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/10208596

**Server:** Internal

**Client/Site:** Portland

**Account:** NA

**Issue:** 

We are experiencing latency when posting checks in SAB. It’s taking approximately 8-20 seconds to advance through each screen which is really slowing the payment processing down. This latency issue started a month or so ago but has been getting progressively worse.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 7, 2019, 09:24

@amishra @smasih @pchaudhary @tim.traylor 

Has there been any changes to the post payments screens in the last few months? I do not believe I have seen any error emails, so I expect something is maybe just not executing as fast as it used to?

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Aug 4, 2019, 23:22

@kdjstudios 


Thank you for the update.

As far as remember there were one little changes on the payment screen page. That ticket was http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1134

Although, there will be no effect on page loading of the payment screen.

**We have checked payment screen page and It's taking approximately 2-3 seconds.**



Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Aug 4, 2019, 23:23

assigned to @amishra

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 5, 2019, 15:02

Marking for discussion as the client is reports 20 second times, while we are reporting 3 second times. There seems to be a difference.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Aug 5, 2019, 23:15

@kdjstudios 

Here is the short video on the payment page.Please have a look.

 * Manual Payment Page- [link](https://www.loom.com/share/3a74eb6e7c07449b940ea41eb9080ced)  
 * Credit Card Page - [Link](https://www.loom.com/share/5d15fad33a3f438fa1abb8d3db28dab2)  

*Might be the network issue in that time while the client had made a payment.*  

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Aug 6, 2019, 02:18

added 23m of time spent

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Aug 8, 2019, 24:16

added 12m of time spent

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 24, 2019, 10:03

I wrote:

> We are wanting to follow up to confirm if we may close this ticket? Over the last few months, the SAB support team has made various updates to SAB CC Processing to help improve it's functionality and speed; while also our networking team has made substantial updates to firewalls and network equipment.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 24, 2019, 11:14

Grant wrote:

> You can close it but it is still latent when processing payments.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 24, 2019, 11:14

closed

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 16, 2020, 09:50

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 16, 2020, 09:50

reopened

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 16, 2020, 09:51

@jredfield @amishra @smasih 

Aavaz, we are reopening this ticket as the site is notifying us that they are still experiencing issues. One of the things I thought of that might be causing this, was that we will attempt to reprocess a payment 3 times, correct? Would that maybe be the scenario this is taking so long, If so, how long should operations expect to wait? Or what other things would cause delays in payments to take 2 minutes overall to post a payment.  

Grant wrote:

> Portland is continuing to experience severe latency issues with SAB. This is especially prominent when processing payments or navigate payment history where it can take 15-30 seconds to advance to the next screen. It takes us on average to 1.75-2mins to post a check payment.

Terri wrote:

> Based upon our call yesterday, I will take this information back to the SAB development team to review.  In the meantime, will you or Jeffrey reach out to us with any date and approximate time frame in which the lag time is 30 seconds or longer please?

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 21, 2020, 06:40

@kdjstudios @jredfield 

Thank you for the update

> Aavaz, we are reopening this ticket as the site is notifying us that they are still experiencing issues. One of the things I thought of that might be causing this, was that we will attempt to reprocess a payment 3 times, correct? Would that maybe be the scenario this is taking so long, If so, how long should operations expect to wait? Or what other things would cause delays in payments to take 2 minutes overall to post a payment.


* Yes, might be causing this issue while reprocessing payment 3 times.or 
* The internet may slow and cause this issue when making a payment.

* Please let us know what do we need to do.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 21, 2020, 09:21

@amishra 

Please only look into what would cause SA Billing to take so long to process the payment. 

We will have our infrastructure team look into other aspects (internet, PC, etc.).

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 31, 2020, 02:47

@kdjstudios @jredfield 


Thank you for the update.

I believe this issue related to the same ticket #1728 .

So that we can mark as child ticket.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @jredfield on Jan 31, 2020, 05:54

@amishra Yes make #1728 the parent ticket

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 31, 2020, 12:50

@amishra @jredfield 

No these are two different tickets that need to be tracked separately. This one is specifically for the payments and delay. The #1728, is in regards to the delay that started after we had to take SAB offline and also the switch to the new LDAP.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Feb 4, 2020, 24:04

@kdjstudios @jredfield 


Thank you for the update.

* I would like to inform you that if the LDAP server is down it may be affected on the payment screen as well. 

* Maybe, Payment Gateway server is down, but we can not test this because I don't figure out when they go down. (It's a very rare case)

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 4, 2020, 13:27

@amishra @smasih @ranand 

Please explain in detail the full payment process. Please provide in detail the longest SAB would take to process a payment (Both Echeck and CC). Please keep in mind the retries, gateway delays/response, etc.

Do NOT consider anything with LDAP. This is only about payment processing times.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Feb 11, 2020, 23:58

@kdjstudios @jredfield 

Thank you for the update.

I would like to inform you that we had already informed you how the payment process works in [comment](http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1482#note_44138) on ticket #1482 

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 12, 2020, 14:18

@amishra @jredfield 

Thanks for the link to the other post. However, I we are needing to know how long each of those steps take, since we are needing to know the processing time for each of those steps.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Feb 13, 2020, 04:53

@kdjstudios @jredfield 

Thank you for the update.

- Credit Card  - Authorize (**~3-4 sec**)
- CCOf         - Authorize CIM  (**~3-3.5 sec**)

- Credit Card  - Moneris (**~3-4 sec**)
- CCOF         -  Moneris CIM  (**~3-4 sec**)

- ECheck       - Authorize (**~3-4 sec**)

- ECheck       - Vericheck (**~3-4 sec**) 

**In short, all payment gateways take max 5 sec to make a successful payment.**


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 13, 2020, 08:21

@amishra @jredfield 

I am so confused, as I would expect that to be the fastest a transaction would take. However, we are asking how long do we expect the transaction may take(Longest)?

Let me provide an example scenario, the user attempts to process a CC transaction and we are not getting a response back from the gateway, so I presume we end up having to process the transaction for the 3 retries, correct?  How long do we wait until we determine the transaction needs to be retried?(10 seconds, 30 seconds, etc.?) So how long do we wait in-between each retry?

* In short, all payment gateways take max 5 sec to make a successful payment.
> How is operations experiencing almost 2 minutes time to process a transaction. http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1499#note_54479

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Mar 31, 2020, 05:42

@kdjstudios @tim.traylor 


Thank you for the update.

* We would like to inform you that we don't know when the server goes down and can not get the response back from the gateway.

* If the user makes  CC transaction and did not get the response from the gateway server they retry after 1 sec.

> How is operations experiencing almost 2 minutes time to process a transaction. http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1499#note_54479

* It might be the network issue.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 31, 2020, 15:08

I wrote:

> Over the last few months there has been multiple updates to Answernet's infrastructure and networking. We are following up to see if your site is still experiencing any delay while processing payments.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 15, 2020, 09:15

I wrote:

> We will be closing this as there have been no reports of latency while posting payments recently. Please let us know if you have any questions or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 15, 2020, 09:15

closed
